app/helpers/samples_helper.rb

```diff
@@ -276,8 +276,8 @@ def report_ready_multiget(pipeline_run_ids)
   end
 
   def top_pipeline_runs_multiget(sample_ids)
-    top_pipeline_runs = PipelineRun.where("id in (select max(id) from pipeline_runs where
-                  sample_id in (#{sample_ids.join(',')}) group by sample_id)")
+    top_pipeline_runs = PipelineRun.where("id in (select x.id from (select max(id) as id from pipeline_runs where
```

Comment:
What does the extra select do in this query?

Reply:
following help from https://www.xaprb.com/blog/2006/04/30/how-to-optimize-subqueries-and-joins-in-mysql/

Results in this improvement:

MySQL [idseq_prod]> explain extended SELECT `pipeline_runs`.* FROM `pipeline_runs` WHERE id in (select x.id from (select max(id) as id from pipeline_runs where                   sample_id in (9151,9144,9143,9142,9141,9140,8461,8460,6979,6978,6977,6545,6544,6543,6192,6174,6173,2319,2318,2317,2316,2315,2314,144,143,142,141,140,139,138) group by sample_id) as x);
+----+-------------+---------------+--------+----------------------------------+----------------------------------+---------+------+------+----------+------------------------------+
| id | select_type | table         | type   | possible_keys                    | key                              | key_len | ref  | rows | filtered | Extra                        |
+----+-------------+---------------+--------+----------------------------------+----------------------------------+---------+------+------+----------+------------------------------+
|  1 | PRIMARY     | <derived3>    | ALL    | NULL                             | NULL                             | NULL    | NULL |   30 |   100.00 | Using where; Start temporary |
|  1 | PRIMARY     | pipeline_runs | eq_ref | PRIMARY                          | PRIMARY                          | 8       | x.id |    1 |   100.00 | End temporary                |
|  3 | DERIVED     | pipeline_runs | range  | index_pipeline_runs_on_sample_id | index_pipeline_runs_on_sample_id | 9       | NULL |   30 |   100.00 | Using where; Using index     |
+----+-------------+---------------+--------+----------------------------------+----------------------------------+---------+------+------+----------+------------------------------+
3 rows in set, 1 warning (0.00 sec)

MySQL [idseq_prod]> explain extended SELECT `pipeline_runs`.* FROM `pipeline_runs` WHERE id in (select max(id) as id from pipeline_runs where                   sample_id in (9151,9144,9143,9142,9141,9140,8461,8460,6979,6978,6977,6545,6544,6543,6192,6174,6173,2319,2318,2317,2316,2315,2314,144,143,142,141,140,139,138) group by sample_id);
+----+-------------+---------------+-------+----------------------------------+----------------------------------+---------+------+-------+----------+--------------------------+
| id | select_type | table         | type  | possible_keys                    | key                              | key_len | ref  | rows  | filtered | Extra                    |
+----+-------------+---------------+-------+----------------------------------+----------------------------------+---------+------+-------+----------+--------------------------+
|  1 | PRIMARY     | pipeline_runs | ALL   | NULL                             | NULL                             | NULL    | NULL | 17239 |   100.00 | Using where              |
|  2 | SUBQUERY    | pipeline_runs | range | index_pipeline_runs_on_sample_id | index_pipeline_runs_on_sample_id | 9       | NULL |    30 |   100.00 | Using where; Using index |
+----+-------------+---------------+-------+----------------------------------+----------------------------------+---------+------+-------+----------+--------------------------+
2 rows in set, 1 warning (0.00 sec)

Reply:
```
MySQL [idseq_prod]> explain extended SELECT `pipeline_runs`.* FROM `pipeline_runs` WHERE id in (select x.id from (select max(id) as id from pipeline_runs where                   sample_id in (9151,9144,9143,9142,9141,9140,8461,8460,6979,6978,6977,6545,6544,6543,6192,6174,6173,2319,2318,2317,2316,2315,2314,144,143,142,141,140,139,138) group by sample_id) as x);
+----+-------------+---------------+--------+----------------------------------+----------------------------------+---------+------+------+----------+------------------------------+
| id | select_type | table         | type   | possible_keys                    | key                              | key_len | ref  | rows | filtered | Extra                        |
+----+-------------+---------------+--------+----------------------------------+----------------------------------+---------+------+------+----------+------------------------------+
|  1 | PRIMARY     | <derived3>    | ALL    | NULL                             | NULL                             | NULL    | NULL |   30 |   100.00 | Using where; Start temporary |
|  1 | PRIMARY     | pipeline_runs | eq_ref | PRIMARY                          | PRIMARY                          | 8       | x.id |    1 |   100.00 | End temporary                |
|  3 | DERIVED     | pipeline_runs | range  | index_pipeline_runs_on_sample_id | index_pipeline_runs_on_sample_id | 9       | NULL |   30 |   100.00 | Using where; Using index     |
+----+-------------+---------------+--------+----------------------------------+----------------------------------+---------+------+------+----------+------------------------------+
3 rows in set, 1 warning (0.00 sec)

MySQL [idseq_prod]> explain extended SELECT `pipeline_runs`.* FROM `pipeline_runs` WHERE id in (select max(id) as id from pipeline_runs where                   sample_id in (9151,9144,9143,9142,9141,9140,8461,8460,6979,6978,6977,6545,6544,6543,6192,6174,6173,2319,2318,2317,2316,2315,2314,144,143,142,141,140,139,138) group by sample_id);
+----+-------------+---------------+-------+----------------------------------+----------------------------------+---------+------+-------+----------+--------------------------+
| id | select_type | table         | type  | possible_keys                    | key                              | key_len | ref  | rows  | filtered | Extra                    |
+----+-------------+---------------+-------+----------------------------------+----------------------------------+---------+------+-------+----------+--------------------------+
|  1 | PRIMARY     | pipeline_runs | ALL   | NULL                             | NULL                             | NULL    | NULL | 17239 |   100.00 | Using where              |
|  2 | SUBQUERY    | pipeline_runs | range | index_pipeline_runs_on_sample_id | index_pipeline_runs_on_sample_id | 9       | NULL |    30 |   100.00 | Using where; Using index |
+----+-------------+---------------+-------+----------------------------------+----------------------------------+---------+------+-------+----------+--------------------------+
2 rows in set, 1 warning (0.00 sec)

```